src/mime.cc

```diff
@@ -262,70 +259,73 @@ mimeInit(char *filename)
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            xstrncpy(chopbuf, buf, BUFSIZ);
 
-        if ((icon = strtok(nullptr, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            const auto pattern = SBuf(strtok(chopbuf, w_space));
+            if (pattern.isEmpty()) {
+                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
+                continue;
+            }
 
-        if ((encoding = strtok(nullptr, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            if ((type = strtok(nullptr, w_space)) == nullptr) {
+                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
+                continue;
+            }
 
-        if ((mode = strtok(nullptr, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            if ((icon = strtok(nullptr, w_space)) == nullptr) {
+                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
+                continue;
+            }
 
-        download_option = 0;
-        view_option = 0;
+            if ((encoding = strtok(nullptr, w_space)) == nullptr) {
+                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
+                continue;
+            }
 
-        while ((option = strtok(nullptr, w_space)) != nullptr) {
-            if (!strcmp(option, "+download"))
-                download_option = 1;
-            else if (!strcmp(option, "+view"))
-                view_option = 1;
-            else
-                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: unknown option: '" << buf << "' (" << option << ")");
-        }
+            if ((mode = strtok(nullptr, w_space)) == nullptr) {
+                debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
+                continue;
+            }
 
-        if (regcomp(&re, pattern, re_flags) != 0) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: regcomp failure: '" << buf << "'");
-            continue;
-        }
+            download_option = 0;
+            view_option = 0;
+
+            while ((option = strtok(nullptr, w_space)) != nullptr) {
+                if (!strcmp(option, "+download"))
+                    download_option = 1;
+                else if (!strcmp(option, "+view"))
+                    view_option = 1;
+                else
+                    debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: unknown option: '" << buf << "' (" << option << ")");
+            }
 
-        m = new MimeEntry(pattern,re,type,encoding,mode,view_option,
```

Comment:
Please continue to _compile_ the pattern here, where we parse/isolate it. In other words, pass/move a RegexPattern object into MimeEntry constructor instead of passing `pattern`. Doing so will allow us to extend/improve the pattern configuration syntax without bothering the MimeEntry class that does not have access to this parsing context. For example, imagine adding a new regex configuration syntax that specifies the regex syntax to be used for the being-configured regex. We want to place that future code somewhere here rather than in MimeEntry constructor...

Doing so would require enabling move support in RegexPattern, which we can easily do after #1232.